src/Service/Engine.php

```diff
@@ -100,14 +101,20 @@ public function dispatchEvent(string $name, DatabaseEvent $event): void
      * @param  string   $message message
      * @param  array    $context optional context data
      */
-    public function log(string $level, string $message, array $context = []): void
+    public function log(string $level, string $message, array $context = [],string $channel = LoggerChannel::__DEFAULT): void
     {
         if (!$this->logger) {
             return;
         }
-
+        Assertion::keyExists($this->logger->loggers, $channel, "Unsupported channel '$channel'");
         $context['database_root'] = (string)$this->database->getPath();
-        $this->logger->log($level, $message, $context);
+        $this->logger->loggers[LoggerChannel::__DEFAULT]->log($level, $message, $context);
+        if ($channel === LoggerChannel::__DEFAULT) {
+            return;
+        }
+        // Log in channel specific logger
+        $this->logger->loggers[$channel]->log($level, $message, $context);
+
```

Comment:
The logging logic/implementation should really not be in the engine, it is not it's responsibility. It shouldn't know how to log, instead, it should only know that it is possible to log by calling the Logger.
The logger should contain the implementation details.
Therefore it would be better to implement the log method in the `logToChannel` method of the logger I discussed earlier.

Also since LoggerChannel is now a type that handles validation internally, we could get rid of the Assertion call you made.
By doing this log(string $level, string $message, array $context = [], ?LoggerChannel $channel = null)
Unfortunately PHP does not allow custom default argument values for cutom types.
To circumvent this, usually, the following strategy is used:
set a default argument value of null
and then check for null in the method

```php
if(!$channel) $channel = new LoggerChannel(LoggerChannel::__DEFAULT);
```
(again this code should go in the logger, not the engine)